### ISSUE TITLE
Implement snapshots functions for CephFS (like RBD ones)

### ISSUE THREAD

**Issue by socketpair** [ceph/ceph-csi#246]
Implement snapshots functions for CephFS (like RBD ones)

**Comment by socketpair** [comment]
@vm86

**Comment by rootfs** [comment]
cc @gman0

**Comment by mykaul** [comment]
@ShyamsundarR - do we know what the gap is?

**Comment by ShyamsundarR** [comment]
@mykaul I do not know this yet, been focusing on the rbd parts. Tagging @ajarr for any comments.

**Comment by mykaul** [comment]
Any updates? @humblec  ?

**Comment by humblec** [comment]
@mykaul sure, that said, at the moment, we are using subdir based cephfs volumes in CSI or its the method followed since start.  Last meeting, we have decided to move on from this method and stick to ceph manager based volume provisioning. 
https://github.com/ceph/ceph-csi/issues/328

Volume provisioning and deletion is planned with v 1.1 release. Once its in, I will implement the snapshot for cephfs. Thats the current plan.

**Comment by mykaul** [comment]
@humblec  - status?

**Comment by humblec** [comment]
@mykaul the snapshot has to wait for ceph manager based volume provisioning and deletion work (https://github.com/ceph/ceph-csi/pull/400)  and also partially on https://github.com/ceph/ceph-csi/issues/411. There is also a design discussion happening to have a common interface to support snapshot and its subsequent operations based on https://github.com/ceph/ceph-csi/issues/411#issuecomment-500490441. 
In short, the preliminary work for supporting this functionality is happening , but we are not there yet completely. May be @batrick can comment more on its preparation from cephfs manager stack/api.

**Comment by batrick** [comment]
The `ceph fs volume` interface has everything needed (imminent v14.2.2) for ceph-csi except the clone-from-snapshot feature (#411). That work is planned for the next sprint. Asynchronous purge is in the current sprint but not a blocker.

**Comment by socketpair** [comment]
https://docs.ceph.com/docs/master/cephfs/fs-volumes/

**Comment by Madhu-1** [comment]
closing this one as its covered in https://github.com/ceph/ceph-csi/issues/702
